lib/web/terminal.go

```diff
@@ -1013,17 +1044,15 @@ func (t *TerminalStream) Write(data []byte) (n int, err error) {
 	return len(data), nil
 }
 
-// Read unwraps the envelope and either fills out the passed in bytes or
-// performs an action on the connection (sending window-change request).
-func (t *TerminalStream) Read(out []byte) (n int, err error) {
+func (t *WsStream) readMessage(out []byte) (string, []byte, int, error) {
```

Comment:
Let's add a godoc specifying what the outputs are since they aren't named and one cannot easily tell what they mean from the signature alone.